rust-src/id/src/ffi.rs

```diff
@@ -292,6 +299,294 @@ pub extern "C" fn ip_info_cdi_verify_key(
     ptr
 }
 
+/// Attempt to create an [`ArInfo`] instance from byte array pointers.
+/// Returns a raw pointer to an [`ArInfo`] instance if
+///  - `public_key_ptr` points to a serialized `elgamal::PublicKey<G1>`
+///    instance,
+///  - `name_ptr`, `url_ptr` and `desc_ptr` to serialized utf8 strings, and
+/// each [`size_t`] parameter with the the `_len` suffix holds the length of
+/// its correspondingly named byte array. Returns a null-pointer if a parameter
+/// could not be deserialized.
+#[no_mangle]
+unsafe extern "C" fn ar_info_create(
+    identity: u32,
+    public_key_ptr: *mut u8,
+    public_key_len: size_t,
+    name_ptr: *const u8,
+    name_len: size_t,
+    url_ptr: *const u8,
+    url_len: size_t,
+    desc_ptr: *const u8,
+    desc_len: size_t,
+) -> *mut ArInfo<G1> {
+    if let Ok(ptr) = ar_info_create_helper(
+        identity,
+        public_key_ptr,
+        public_key_len,
+        name_ptr,
+        name_len,
+        url_ptr,
+        url_len,
+        desc_ptr,
+        desc_len,
+    ) {
+        ptr
+    } else {
+        std::ptr::null_mut()
+    }
+}
+
+/// Attempt to create an [`ArInfo`] instance from byte array pointers.
+/// Returns [`Ok`] if
+///  - `public_key_ptr` points to a serialized `elgamal::PublicKey<G1>`
+///    instance,
+///  - `name_ptr`, `url_ptr` and `desc_ptr` to serialized utf8 strings, and
```

Comment:
There isn't any serialization involved here. `name_ptr`, `name_len` must point to a valid utf8 encoded string.